manifests/functions/create_export.pp

```diff
@@ -34,13 +39,16 @@
       content => $line,
     }
 
+    # Create the directory path only if a File resource isn't
+    #     # defined previously AND the $create_dir boolean is true.
     unless defined(File[$name]) {
-      file { $name:
-        ensure                  => directory,
-        owner                   => $owner,
-        group                   => $group,
-        mode                    => $mode,
-        selinux_ignore_defaults => true,
+      if $create_dir {
+        file { $name:
+          ensure => directory,
+          owner  => $owner,
+          group  => $group,
+          mode   => $mode,
+        }
```

Comment:
Given it will not manage the resource, I suggest to name the new param `manage` rather than `create` (i.e. the file resource create a non-existing directory but also ensure owner, group, permissions, etc).

Reply:
If $create_dir is true (which is the default) then the directory is managed.  Or am misunderstanding your comment?  My use case here is that I have a 3rd party application that expects the directory to be absent.  The application install will fail if the directory is created but I still have to export the directory via NFS.  This solves the problem of exporting the directory but not being required to create the directory/export.

Reply:
Sorry to be unclear, I am suggesting naming the parameter `manage_dir` rather than `create_dir`.

Or `manage_directory` as bytes are cheap nowadays :smile: 

Because if you set `manage_directory => false` and you put the wrong owner / group on the directory you created by hand, puppet will not fix this because it does not manage that directory.

Reply:
Changed per your recommendations.  Thanks.